docs/dapp/sapphire/precompiles.md

```diff
@@ -238,3 +255,70 @@ bytes memory digest = abi.encodePacked(keccak256("signed message"));
 bytes memory signature = Sapphire.sign(alg, sk, digest, "");
 require( Sapphire.verify(alg, pk, digest, "", signature) );
 ```
+
+## SHA-512
+
+### Overview
+
+ * Precompile address: `0x0100000000000000000000000000000000000101`
+ * Parameters: `bytes input_data`
+
+Hash the input data with SHA-512, according to [NIST.FIPS.180-4]
+
+:::note SHA-512 is vulnerable to length-extension attacks
+
+Which is relevant if you are computing the hash of a secret message. The SHA-512/256 variant is not vulnerable to length-extension attacks.
+
+:::
+
+### Gas Cost
+
+* 115 gas, then 13 gas per word
+
+### Example
+
+```solidity
+bytes memory result = sha512(abi.encodePacked("input data"));
+```
+
+
+## SHA-512/256
+
+### Overview
+
+ * Precompile address: `0x0100000000000000000000000000000000000102`
+ * Parameters: `bytes input_data`
+
+Hash the input data with SHA-512/256, according to [NIST.FIPS.180-4]
+
+[NIST.FIPS.180-4]: https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
+
+### Gas Cost
+
+ * 115 gas, then 13 gas per word
+
+### Example
+
+```solidity
+bytes32 result = sha512_256(abi.encodePacked("input data"));
+```
+
+
+## Subcall
+
+### Overview
+
+ * Precompile address: `0x0100000000000000000000000000000000000102`
+ * Parameters: `string method, bytes cborEncodedParams`
+
+Performs an Oasis SDK call, which allows Sapphire contracts to interact with the Consensus layer and other modules supported by the SDK. For more information about the specific modules and their available queries or calls see the [Oasis SDK modules source].
```

Comment:
(nit) Matevz will probably ask for this to be split up.

Reply:
The sentences are below 25 words each, it is a paragraph.